### PR TITLE
combine the xref spec builder and json schema xref loader

### PR DESCRIPTION
**Pull request by OsmondJiang**
1. no need sort xref specs built from sdp
2. reduce un-necessary inner function 
3. combine internal xref spec build logic with json schema xref loader

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/docfx/pull/4864)

### CODE REVIEW

File: src/docfx/build/xref/IXrefSpec.cs [hunk 1]
```diff
@@ -7,6 +7,8 @@ namespace Microsoft.Docs.Build
 {
     internal interface IXrefSpec
     {
+        SourceInfo Source { get; }
```
Comment (yufeih) [inline src/docfx/build/xref/IXrefSpec.cs]: is adding source info to `InternalXrefSpec` is enough?

File: src/docfx/lib/schema/JsonSchemaTransformer.cs [hunk 2]
```diff
@@ -5,7 +5,9 @@
 using System.Collections.Concurrent;
 using System.Collections.Generic;
 using System.Linq;
+using System.Text.RegularExpressions;
 using System.Threading;
+using Microsoft.Graph;
```
Comment (yufeih) [inline src/docfx/lib/schema/JsonSchemaTransformer.cs]: remove this?

Comment (OsmondJiang) [reply src/docfx/lib/schema/JsonSchemaTransformer.cs]: my IDE will automatically add some strange namespace... :(

Comment (yufeih) [reply src/docfx/lib/schema/JsonSchemaTransformer.cs]: me too, it is a new annoying IDE feature

**Pull request merged**